blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -872,3 +782,123 @@ Blockly.ContractEditor.prototype.changeParameterName_ = function(paramID, newNam
   var oldName = paramInfo.name;
   Blockly.Variables.renameVariable(oldName, newName, Blockly.modalBlockSpace)
 };
+
+/**
+ * Go through each of our example blocks and figure out which is widest
+ */
+Blockly.ContractEditor.prototype.getMaxExampleBlockWidth_ = function () {
+  return this.exampleBlocks.reduce(function (previousMax, block) {
+    var functionCallBlock = block.getInputTargetBlock(
+      Blockly.ContractEditor.EXAMPLE_BLOCK_ACTUAL_INPUT_NAME);
+    if (!functionCallBlock) {
+      return previousMax;
+    }
+    var width = functionCallBlock.getHeightWidth().width;
+    return Math.max(previousMax, width);
+  }, 0);
+};
+
+/**
+ * Call reset on each of our views
+ */
+Blockly.ContractEditor.prototype.resetExampleViews = function () {
+  this.exampleViews_.forEach(function (exampleView) {
+    exampleView.reset();
+  });
+};
+
+/**
+ * Call our app-specific test handler for this block
+ */
+Blockly.ContractEditor.prototype.testExample = function (block) {
+  return this.testHandler_(block);
+};
+
+/**
+ * Call our app-specific test reset handler for this block
+ */
```

Comment:
Reset doesn't return anything as far as I can tell (only used by eval). However, I do want this for testExample which returns a string.